web/src/read_receipts.ts

```diff
@@ -15,91 +15,111 @@ import * as people from "./people.ts";
 import * as ui_report from "./ui_report.ts";
 import * as util from "./util.ts";
 
+const read_receipts_polling_interval_ms = 60 * 1000;
 const read_receipts_api_response_schema = z.object({
     user_ids: z.array(z.number()),
 });
 
-export function show_user_list(message_id: number): void {
-    $("#read-receipts-modal-container").html(render_read_receipts_modal({message_id}));
-    modals.open("read_receipts_modal", {
-        autoremove: true,
-        on_shown() {
-            const message = message_store.get(message_id);
-            assert(message !== undefined, "message is undefined");
+let interval_id: number | null = null;
+let is_read_list_visible = false;
+
+export function fetch_read_receipts(message_id: number): void {
+    const message = message_store.get(message_id);
+    assert(message !== undefined, "message is undefined");
+
+    if (message.sender_email === "notification-bot@zulip.com") {
+        $("#read_receipts_modal .read_receipts_info").text(
+            $t({
+                defaultMessage: "Read receipts are not available for Notification Bot messages.",
+            }),
+        );
+        $("#read_receipts_modal .modal__content").addClass("compact");
+        return;
+    }
 
-            if (message.sender_email === "notification-bot@zulip.com") {
+    if (!is_read_list_visible) {
+        loading.make_indicator($("#read_receipts_modal .loading_indicator"));
+    }
+
+    void channel.get({
+        url: `/json/messages/${message_id}/read_receipts`,
+        success(raw_data) {
+            const $modal = $("#read_receipts_modal").filter(`[data-message-id=${message_id}]`);
+            // If the read receipts modal for the selected message ID is closed
+            // by the time we receive the response, return immediately.
+            if ($modal.length === 0) {
+                return;
+            }
+
+            $("#read_receipts_modal .read_receipts_error").text("");
```

Comment:
This does not appear to be a correct way to hide an error banner element. Changed to the following:

``` diff
diff --git a/web/src/read_receipts.ts b/web/src/read_receipts.ts
index 784ebac9ce..de498e59a2 100644
--- a/web/src/read_receipts.ts
+++ b/web/src/read_receipts.ts
@@ -52,7 +52,7 @@ export function fetch_read_receipts(message_id: number): void {
             }
 
             has_initial_data = true;
-            $("#read_receipts_modal .read_receipts_error").text("");
+            $("#read_receipts_modal .read_receipts_error").removeClass("show");
             const data = read_receipts_api_response_schema.parse(raw_data);
             const users = data.user_ids.map((id) => people.get_user_by_id_assert_valid(id));
             users.sort(people.compare_by_name);
```